src/main/java/core/basesyntax/FigureSupplier.java

```diff
@@ -0,0 +1,22 @@
+package core.basesyntax;
+
+import java.util.Random;
+
+public class FigureSupplier {
+    private final ColorSupplier colorSupplier = new ColorSupplier();
+    private final Random random = new Random();
+
+    public Figure getRandomFigure() {
+        int anInt = random.nextInt(5);
+        return switch (anInt) {
+            case 1 -> new Circle(Colors.WHITE.name(), 10);
+            case 2 -> new IsoscelesTrapezoid(colorSupplier.getRandomColor(),
+                    random.nextInt(10), random.nextInt(10));
```

Comment:
Avoid passing expressions directly in the constructor. Instead, create separate variables for the parameters and pass them to the constructor for better code readability.